### ISSUE TITLE
Image undefined until verification fails

### ISSUE THREAD

**Issue by CarlSinclair** [zpfz/RVerify.js#2]
![image](https://user-images.githubusercontent.com/69092633/90966781-1d48b200-e4ad-11ea-864a-9fe78efc6183.png)

After verification fails once, the image appears properly.

**Comment by CarlSinclair** [comment]
Can I start it with code "0" to make it think it already failed?

**Comment by zpfz** [comment]
You need to set the image to be used for verification via the property `album`，you can see here to get more details: [album](https://rverify.seeyoz.cn/#album)

**Comment by zpfz** [comment]
What's more, to ensure that images load first, add the code like this in the header:
```html
<link rel="preload" href="YOUR_IMAGE_PATH" as="image">
```
or use CDN resource links to solve.

**Comment by CarlSinclair** [comment]
I have the image set via the `album` property, but it only appears after the first failure.

Preloading it didn't help.

**Comment by zpfz** [comment]
If you can, please use [codepen.io](codepen.io) to reproduce the problem, my preliminary judgment is that the image loading is too slow to cause, you can try to change the access speed of the image, find some pictures to test from `Bing`.

**Comment by CarlSinclair** [comment]
I can't get it to work at all on codepen.
![image](https://user-images.githubusercontent.com/69092633/91072565-16868000-e610-11ea-9d51-27a96e2a699e.png)

**Comment by CarlSinclair** [comment]
I really don't think this is related to speed. Here's a demonstration of the issue: [imgur.com/a/5eyTCe4](https://imgur.com/a/5eyTCe4)

**Comment by zpfz** [comment]
> I can't get it to work at all on codepen.
> ![image](https://user-images.githubusercontent.com/69092633/91072565-16868000-e610-11ea-9d51-27a96e2a699e.png)

The way you're using the rverify plugin doesn't seem quite right. You can access the codepen example  directly what I created( [https://codepen.io/zpfz/pen/MWypYKd](https://codepen.io/zpfz/pen/MWypYKd) ), or follow the image below to use the links to the corresponding resource:
![dcFeaR.png](https://s1.ax1x.com/2020/08/25/dcFeaR.png)
![dcFZZ9.png](https://s1.ax1x.com/2020/08/25/dcFZZ9.png)

**Comment by zpfz** [comment]
I'm so sorry that I can't reproduce your feedback on my end. But I tested pretty well with `codepen.io` and `codesandbox.io`. You can see here with demo: https://codepen.io/zpfz/pen/MWypYKd and https://codesandbox.io/s/rverifyjs-demo-g3iol (the image resource links from `Bing`). And you can also add the following code to the `<head></head>` (If the gallery has the anti-theft chain turned on)：
```html
<meta name="referrer" content="never">
```
![dcFAr4.gif](https://s1.ax1x.com/2020/08/25/dcFAr4.gif)
You can test first: RVerify docs's homepage has a `TRY ONLINE` button, click it will popup a demo and check it out whether has the similar problem. after all, homepage's demo image resources from `vercel.com`, loading speed is also very fast!
Or you can just use the image resource links of the code example from `album` in the RVerify's docs.
![dckNm4.png](https://s1.ax1x.com/2020/08/25/dckNm4.png)
You can also paste the links to your album, I will try to access it here, but I hope I can access it without the VPN.

**Comment by zpfz** [comment]
By the way, if it's convenient, please paste the code you used to configure RVerify, something like this（especially the album part, I'd like to know if there is a problem with the source of the images）:
```js
RVerify.configure({
  ...
})
```

**Comment by CarlSinclair** [comment]
Yes, I'm using it differently. I'm not putting it behind a button. I'm making the verification box appear as soon as the page loads. Maybe this is the problem?

Here's my very simple test config:
```
<script defer>
RVerify.action(function(res){
  console.log(res);
});
RVerify.configure({
  album: ['/themes/New/img/right-round.jpg',]
  
});
</script>
```

Is it not possible to have it in-line in the form without needing to click a button to make it appear?

**Comment by zpfz** [comment]
Oh, I think I know what the problem is, based on what you've described. It's necessary for you to load the images finish before when RVerify was used. Of course, it's possible to have it in-line in the form directly. So you need to find a way to load the image resources finish before using RVerify. I have two ways, you can refer to it:
1.Via CSS: Batch load images via the `background-image` property.
```html
<div id="preload" style="display:none"></div>
```
```css
#preload{
　　// The images in the url will be requested in turn.
    background:url('./images/goods1.png');
    background:url('./images/goods2.png');
    background:url('./images/goods3.png');
}
```
2.Via JS: batch create Image objects, assign their `src` property to the URL of the image we want to load.
```js
RVerify.configure({
  ...
});
// List of URLs to load images
var imgList = [
    './images/goods1.png',
    './images/goods2.png'
];
loadImg(imgList);
function loadImg(imgList){
    imgList.map(function(imgSrc,index){
        var img = new Image();
        // Record the number of loaded images
        img.num = index+1;
        console.log(img.num);
        // Binding load completion event handling functions
        img.addEventListener("load",loadHandler);
        // After specifying src for the img, the browser will send a request
        img.src = imgSrc;
    })
}
function loadHandler(e){
    // Logout event to free memory
    this.removeEventListener("load",loadHandler);
    // Determine if it is the last image to be loaded
    if(this.num === imgList.length){
        console.log("Load finished!");
        RVerify.action(function(res){
            console.log(res);
        });
    }
}
```
I have to go to bed. Good night~

**Comment by CarlSinclair** [comment]
Goodnight.

Have you tried the CSS method you suggested? I prefer CSS because it's lighter than JS and won't interfere with my other JS.

Sadly it did not work for me. I even tried to make the image visible. It appears, but the spinning image (same image) is still "undefined" until after first attempt.
![image](https://user-images.githubusercontent.com/69092633/91211234-d47b3e00-e6e4-11ea-971c-8a4c2a39274e.png)

**Comment by zpfz** [comment]
In fact, there are too many uncertainties to solve with CSS alone, I still recommend using JS, I will optimize the problem in RVerify, but recently more busy, I will have time after Sep. 5th, if you are in a hurry to solve this problem, you can use the Google search "js how to determine the image loading is complete"(All in all, let the images load before using RVerify).

**Comment by CarlSinclair** [comment]
Thank you. It's fine, I'm not in a hurry. I can wait 😄 Thank you for the great work.

**Comment by zpfz** [comment]
Hey~bro, I'm coming🍻! A question just occurred to me, did you put `RVerify.action` before `RVerify.configure`, causing `RVerify.action` to execute first?

**Comment by zpfz** [comment]
I have released version `0.1.1`, enjoy it~
